cpp/plugins/cucim.kit.cuslide/src/cuslide/jpeg/libjpeg_turbo.cpp

```diff
@@ -196,6 +231,119 @@ bool read_jpeg_header_tables(const void* handle, const void* jpeg_buf, unsigned
     return true;
 }
 
+// tjDecompress2
+int jpeg_decode_buffer(const void* handle,
+                       const unsigned char* jpegBuf,
+                       unsigned long jpegSize,
+                       unsigned char* dstBuf,
+                       int width,
+                       int pitch,
+                       int height,
+                       int pixelFormat,
+                       int flags,
+                       int jpegColorSpace)
+{
+    JSAMPROW* row_pointer = NULL;
+    int i, retval = 0, jpegwidth, jpegheight, scaledw, scaledh;
+
+    tjinstance* instance = (tjinstance*)handle;
+    j_decompress_ptr dinfo = NULL;
+    dinfo = &instance->dinfo;
+    instance->jerr.warning = FALSE;
+    instance->isInstanceError = FALSE;
+
+    instance->jerr.stopOnWarning = (flags & TJFLAG_STOPONWARNING) ? TRUE : FALSE;
+    if ((instance->init & DECOMPRESS) == 0)
+        THROW("tjDecompress2(): Instance has not been initialized for decompression");
+
+    if (jpegBuf == NULL || jpegSize <= 0 || dstBuf == NULL || width < 0 || pitch < 0 || height < 0 || pixelFormat < 0 ||
+        pixelFormat >= TJ_NUMPF)
+        THROW("tjDecompress2(): Invalid argument");
+
+#ifndef NO_PUTENV
+    if (flags & TJFLAG_FORCEMMX)
+        putenv("JSIMD_FORCEMMX=1");
+    else if (flags & TJFLAG_FORCESSE)
+        putenv("JSIMD_FORCESSE=1");
+    else if (flags & TJFLAG_FORCESSE2)
+        putenv("JSIMD_FORCESSE2=1");
+#endif
+
+    if (setjmp(instance->jerr.setjmp_buffer))
+    {
+        /* If we get here, the JPEG code has signaled an error. */
+        retval = -1;
+        goto bailout;
+    }
+
+    jpeg_mem_src_tj(dinfo, jpegBuf, jpegSize);
+    jpeg_read_header(dinfo, TRUE);
+
+    // Modified to set jpeg_color_space for cuCIM
+    if (jpegColorSpace)
+    {
+        dinfo->jpeg_color_space = static_cast<J_COLOR_SPACE>(jpegColorSpace);
+    }
+
+    instance->dinfo.out_color_space = pf2cs[pixelFormat];
+    if (flags & TJFLAG_FASTDCT)
+        instance->dinfo.dct_method = JDCT_FASTEST;
+    if (flags & TJFLAG_FASTUPSAMPLE)
+        dinfo->do_fancy_upsampling = FALSE;
```

Comment:
We seem to use braces in some places (like above), but not here. Would it make sense to just use braces everywhere?

Reply:
Same reason for above.